nutils/evaluable.py

```diff
@@ -1115,13 +1116,26 @@ def _simplified(self):
         if len(where) < self.ndim - 1:
             return align(Orthonormal(basis, vector), (*where, self.ndim - 1), self.shape)
 
-    @staticmethod
-    def evalf(G, n):
-        GG = numpy.einsum('...ki,...kj->...ij', G, G)
-        v1 = numpy.einsum('...ij,...i->...j', G, n)
-        v2 = numpy.linalg.solve(GG, v1)
-        v3 = numpy.einsum('...ij,...j->...i', G, v2)
-        return numeric.normalize(n - v3)
+    if LooseVersion(numpy.version.version) < LooseVersion('1.22'):
+
+        @staticmethod
+        def evalf(G, n):
+            GG = numpy.einsum('...ki,...kj->...ij', G, G)
+            v1 = numpy.einsum('...ij,...i->...j', G, n)
+            v2 = numpy.linalg.solve(GG, v1)
+            v3 = numpy.einsum('...ij,...j->...i', G, v2)
+            return numeric.normalize(n - v3)
+
+    else:
+
+        # Faster implementation for Numpy versions that support QR
+        # factorization of arrays with arbitrarily many dimensions.
```

Comment:
With QR we don't get an exception when `G` is linearly dependent. I'm not sure if we want this.

Reply:
Removed it.